joplin/users/forms.py

```diff
@@ -0,0 +1,50 @@
+from django import forms
+from django.utils.translation import ugettext_lazy as _
+from wagtail.users.forms import UserEditForm, UserCreationForm
+from django.core.exceptions import ValidationError
+
+from django.contrib.auth.models import Group
+from groups.models import Department
+
+
+'''
+    Overwrite clean() method for Users.
+    Ensure that non-admins choose both a Role and a Department.
+    Set user-selected "Roles" and "Department" into "Groups".
+'''
+def custom_user_clean(self, cleaned_data):
```

Comment:
Similar to PublishPreflight, we had to modify the clean() function for user form validation. We add extra checks to make sure that if a user isn't a superuser Administrator, then they have both a role and a department